File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_ScratchPrize.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_ScratchPrize implements ITableOperate {
 public final static String TAB_FILE_DATA = "/scratch_prize.txt";
 public static final int ID_HEAD_ICON = 0 ;
public static final int ID_WEIGHT = 1 ;
public static final int MAX_RECORD = 2 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private String m_HeadIcon;
 public String GetHeadIcon() { return m_HeadIcon;}

private int m_Weight;
 public int GetWeight() { return m_Weight;}

public boolean LoadTable(HashMap<Integer, Table_ScratchPrize> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_ScratchPrize> _hash = (HashMap<Integer, Table_ScratchPrize> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_ScratchPrize _values = new Table_ScratchPrize();
 _values.m_HeadIcon =  (String)valuesList.get(ID_HEAD_ICON);
_values.m_Weight =  Integer.parseInt((String)valuesList.get(ID_WEIGHT));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

File: TLBB1.5/tabletools/CodeTable/CSharp/Table_ScratchPrize.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_ScratchPrize : ITableOperate
{ private const string TAB_FILE_DATA = "scratch_prize.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_HEAD_ICON,
ID_WEIGHT,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private string m_HeadIcon;
 public string HeadIcon { get{ return m_HeadIcon;}}

private int m_Weight;
 public int Weight { get{ return m_Weight;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_ScratchPrize _values = new Tab_ScratchPrize();
 _values.m_HeadIcon =  valuesList[(int)_ID.ID_HEAD_ICON] as string;
_values.m_Weight =  Convert.ToInt32(valuesList[(int)_ID.ID_WEIGHT] as string);

 _hash.Add(nKey,_values); }


}
}

